app/services/source_refresh_service.rb

```diff
@@ -0,0 +1,46 @@
+class SourceRefreshService
+  def initialize(source_id)
+    @source = Source.find(source_id)
+  end
+
+  def process
+    if @source.refresh_task_id.nil?
+      dispatch_refresh_upload_task
+    else
+      task = Task.find(@source.refresh_task_id)
+
+      if task.state == "completed"
+        if task.child_task_id.nil?
+          Rails.logger.info("PersisterTask is creating, please try again later")
```

Comment:
@hsong-rh Should this be in a separate function and we can raise 429 from there.

Reply:
@hsong-rh https://github.com/rack/rack/blob/a05f8d56f9ac4da14dddb8f312a3b43644f73397/lib/rack/utils.rb#L569

Reply:
@hsong-rh 
Rack::Utils::SYMBOL_TO_STATUS_CODE[:too_many_requests]